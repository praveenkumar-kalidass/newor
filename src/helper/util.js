import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
