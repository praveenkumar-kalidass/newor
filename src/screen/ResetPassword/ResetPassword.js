import React from 'react';
import { Text } from 'native-base';

const ResetPassword = () => <Text>Reset password</Text>;

export default ResetPassword;
