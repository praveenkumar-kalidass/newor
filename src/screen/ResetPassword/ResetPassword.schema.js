import Joi from 'joi';

export const SCHEMA = Joi.object({
  password: Joi.string().alphanum().min(6).max(15)
    .required(),
  confirmPassword: Joi.any().valid(Joi.ref('password')).required(),
});

export const INITIAL_STATE = {
  password: '',
  confirmPassword: '',
};

export const FIELDS = [
  {
    key: 'password',
    placeholder: 'PASSWORD',
  },
  {
    key: 'confirmPassword',
    placeholder: 'CONFIRM_PASSWORD',
  },
];
