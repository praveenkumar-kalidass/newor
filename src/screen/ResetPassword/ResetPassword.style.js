import styled from 'styled-components/native';
import { Box, Text, Input } from 'native-base';

export const Container = styled(Box).attrs(({ theme }) => ({
  borderRadius: 20,
  flex: 1,
  alignSelf: 'center',
  p: 5,
  bg: theme.color.MODAL,
  elevation: 1,
  shadowColor: theme.color.BACKGROUND_0,
}))``;

export const Title = styled(Text).attrs(() => ({
  bold: true,
  fontSize: 20,
}))``;

export const InputField = styled(Input).attrs(({ isInvalid, theme }) => ({
  _focus: {
    bg: theme.color.BACKGROUND_100,
    borderColor: theme.color.SECONDARY_100,
  },
  mb: isInvalid ? 0 : 5,
}))`
  font-size: 14px;
`;
