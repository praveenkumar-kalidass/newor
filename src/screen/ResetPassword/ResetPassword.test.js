import React from 'react';

import { act, render, fireEvent } from 'test/util';
import ResetPassword from './ResetPassword';

const mockNavigation = {
  navigate: jest.fn(),
};
jest.mock('@react-navigation/native', () => {
  const reactNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...reactNavigation,
    useNavigation: () => mockNavigation,
    useRoute: () => ({ params: 'testtoken123' }),
  };
});
const mockToast = { show: jest.fn() };
jest.mock('native-base', () => {
  const nativeBase = jest.requireActual('native-base');
  return {
    ...nativeBase,
    useToast: () => mockToast,
  };
});
const mockUseUser = {
  resetPassword: jest.fn(),
};
jest.mock('api/useUser', () => () => mockUseUser);

describe('ResetPassword', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', async () => {
    const container = await render(<ResetPassword />);

    expect(container).toMatchSnapshot();
  });

  it('should show toast when reset password fails', async () => {
    mockUseUser.resetPassword.mockRejectedValueOnce({ response: { data: { code: 'NEWOR_INTERNAL_SERVER_ERROR' } } });

    const { getByTestId } = await render(<ResetPassword />);

    await act(async () => {
      await fireEvent.changeText(getByTestId('reset-password-input-password'), 'test1234');
      await fireEvent(getByTestId('reset-password-input-password'), 'onBlur');
      await fireEvent.changeText(getByTestId('reset-password-input-confirmPassword'), 'test1234');
      await fireEvent(getByTestId('reset-password-input-confirmPassword'), 'onBlur');
    });

    await act(async () => {
      await fireEvent.press(getByTestId('reset-password-submit'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(1);
  });

  it('should show modal and navigate when reset password is success', async () => {
    mockUseUser.resetPassword.mockResolvedValueOnce();

    const { getByTestId, queryByTestId } = await render(<ResetPassword />);

    await act(async () => {
      await fireEvent.changeText(getByTestId('reset-password-input-password'), 'test1234');
      await fireEvent(getByTestId('reset-password-input-password'), 'onBlur');
      await fireEvent.changeText(getByTestId('reset-password-input-confirmPassword'), 'test1234');
      await fireEvent(getByTestId('reset-password-input-confirmPassword'), 'onBlur');
    });

    await act(async () => {
      await fireEvent.press(getByTestId('reset-password-submit'));
    });

    expect(queryByTestId('reset-password-success-modal')).not.toBeNull();

    await act(async () => {
      await fireEvent.press(getByTestId('reset-password-success-login'));
    });

    expect(mockNavigation.navigate).toHaveBeenCalledTimes(1);
    expect(mockNavigation.navigate).toHaveBeenCalledWith('LOGIN');
  });
});
