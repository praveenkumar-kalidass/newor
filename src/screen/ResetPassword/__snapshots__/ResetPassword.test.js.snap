// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "position": "absolute",
        },
      ]
    }
  >
    <RNSVGSvgView
      bbHeight={1334}
      bbWidth={750}
      focusable={false}
      height={1334}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 1334,
            "width": 750,
          },
        ]
      }
      width={750}
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M0 200 L0 300 L750 200 L750 100 L0 200 Z"
          fill={4289527808}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M0 350 L0 370 L750 270 L750 250 L0 350 Z"
          fill={4289527808}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M0 1084 L0 1104 L750 1004 L750 984 L0 1084 Z"
          fill={4278199178}
          propList={
            Array [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M0 1154 L0 1254 L750 1154 L750 1054 L0 1154 Z"
          fill={4278199178}
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    flex={1}
    onLayout={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        }
      }
    >
      <View
        dataSet={Object {}}
        elevation={1}
        shadowColor="#000000"
        shadowOffset={
          Object {
            "height": 1,
            "width": 0,
          }
        }
        shadowOpacity={0.8}
        shadowRadius={1}
        style={
          Array [
            Object {
              "alignSelf": "center",
              "backgroundColor": "#eeeeee",
              "borderRadius": 20,
              "flex": 1,
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          style={
            Array [
              Object {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 20,
              },
              Array [
                Object {},
              ],
            ]
          }
          styleProps={
            Object {
              "mb": 5,
            }
          }
        >
          <View
            dataSet={Object {}}
            style={
              Array [
                Object {
                  "backgroundColor": "#00238a",
                  "color": "#00238a",
                  "marginRight": 8,
                  "paddingRight": 8,
                },
                Array [
                  Object {},
                ],
              ]
            }
          />
          <Text
            dataSet={Object {}}
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "letterSpacing": 0,
                  "lineHeight": 30,
                  "textDecorationLine": undefined,
                },
                Array [
                  Object {},
                ],
              ]
            }
          >
            RESET_PASSWORD
          </Text>
        </View>
        <View
          dataSet={Object {}}
          isInvalid=""
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 20,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessibilityDescribedBy="field-react-aria-1-feedback"
              accessible={true}
              dataSet={Object {}}
              disabled={false}
              editable={true}
              isInvalid=""
              isRequired={false}
              nativeID="field-react-aria-1"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="PASSWORD"
              placeholderTextColor="#a3a3a3"
              readOnly={false}
              required={false}
              secureTextEntry={true}
              selectionColor="#1f2937"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                  ],
                ]
              }
              testID="reset-password-input-password"
              value=""
            />
          </View>
        </View>
        <View
          dataSet={Object {}}
          isInvalid=""
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            dataSet={Object {}}
            isFocused={false}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 20,
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessibilityDescribedBy="field-react-aria-2-feedback"
              accessible={true}
              dataSet={Object {}}
              disabled={false}
              editable={true}
              isInvalid=""
              isRequired={false}
              nativeID="field-react-aria-2"
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholder="CONFIRM_PASSWORD"
              placeholderTextColor="#a3a3a3"
              readOnly={false}
              required={false}
              secureTextEntry={true}
              selectionColor="#1f2937"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#171717",
                    "flex": 1,
                    "fontFamily": undefined,
                    "fontSize": 12,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "height": "100%",
                    "paddingBottom": 8,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                    "paddingTop": 8,
                    "width": "100%",
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                    },
                  ],
                ]
              }
              testID="reset-password-input-confirmPassword"
              value=""
            />
          </View>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          colorScheme="primary"
          dataSet={Object {}}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ad0000",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 0.4,
              "paddingBottom": 10,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 10,
            }
          }
          testID="reset-password-submit"
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
            test={true}
          >
            <View
              dataSet={Object {}}
              style={Object {}}
            >
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#fafafa",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                SUBMIT
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      testID="reset-password-success-modal"
      transparent={true}
      visible={true}
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.5)",
              "display": "flex",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          elevation={1}
          shadowColor="#000000"
          style={
            Array [
              Object {
                "alignSelf": "center",
                "backgroundColor": "#eeeeee",
                "borderRadius": 20,
                "flex": 1,
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#171717",
                  "fontFamily": undefined,
                  "fontSize": 20,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "letterSpacing": 0,
                  "lineHeight": 30,
                  "textDecorationLine": undefined,
                }
              }
            >
              SUCCESS
            </Text>
          </View>
          <View
            aria-orientation="horizontal"
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#d4d4d4",
                "height": 1,
                "marginBottom": 8,
                "marginTop": 8,
                "width": "100%",
              }
            }
            thickness="1"
          />
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                accessibilityLabel="RESET_PASSWORD_SUCCESS"
                alt="RESET_PASSWORD_SUCCESS"
                dataSet={Object {}}
                onError={[Function]}
                source={
                  Object {
                    "testUri": "../../../src/asset/image/newor-success.png",
                  }
                }
                style={
                  Object {
                    "height": 256,
                    "maxWidth": "100%",
                    "width": 256,
                  }
                }
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 16,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 24,
                    "textDecorationLine": undefined,
                  }
                }
              >
                RESET_PASSWORD_SUCCESS
              </Text>
            </View>
          </View>
          <View
            aria-orientation="horizontal"
            dataSet={Object {}}
            style={
              Object {
                "backgroundColor": "#d4d4d4",
                "height": 1,
                "marginBottom": 20,
                "marginTop": 20,
                "width": "100%",
              }
            }
            thickness="1"
          />
          <View
            dataSet={Object {}}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "flex-end",
              }
            }
          >
            <View
              accessibilityRole="button"
              accessible={true}
              collapsable={false}
              colorScheme="primary"
              dataSet={Object {}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ad0000",
                  "borderRadius": 4,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 10,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 10,
                }
              }
              testID="reset-password-success-login"
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={Object {}}
                  style={Object {}}
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#fafafa",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    LOGIN
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</RNCSafeAreaProvider>
`;
